### ISSUE TITLE
Adaptive Card Details are read two times by NVDA screen reader.

### ISSUE THREAD

**Issue by PuspanjaliMuduli** [nvaccess/nvda#14448]
**Steps to reproduce:**
1. Enable NVDA Screen Reader.
2. Trigger the feature on the bot.
3. When you get the response, focus on the any adaptive card.
4. Observe the screen reader input.
**Actual behavior:**
When chatting with a chatbot on teams channel While focusing on the adaptive card using short cut keys 
NVDA screen reader reading the card text as 2 times.
**Expected behavior:**
When chatting with a chatbot on teams channel, while we focus on any adaptive card using short cut keys 
it should read card text only once.
NVDA logs, crash dumps and other attachments:
**System configuration**
NVDA installed/portable/running from source:
Windows 10
**NVDA version:**
2022.3.1
Windows version: 10
Name and version of other software in use when reproducing the issue: 2022.3.1
Other information about your system:
N/A
**Other questions:**
We are usnig the MS Adaptive cards which is in JSON format to show the UI design on the bot.
Does the issue still occur after restarting your computer? Yes
Have you tried any other versions of NVDA? If so, please report their behaviors. 
Yes, the issue still exist. NVDA version - 2022.2.2
If NVDA add-ons are disabled, is your problem still occurring? 
Not sure, Please help us to check on this part.
Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
 Not sure, Please help us to check on this part.

--Kindly help on the issue to check & let us know if you need more details on the same.

**Comment by PuspanjaliMuduli** [comment]
https://user-images.githubusercontent.com/109591184/207843608-9b75ba52-f975-437c-8d60-e81674c894b8.mp4

Have also attached the document for your reference.

**Comment by seanbudd** [comment]
It is still not clear what the software is that you are using here.
Is this a website you created yourself?
To be able to reproduce and investigate this issue, we need a minimal code sample.
Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/) which reproduces the issue.

**Comment by PuspanjaliMuduli** [comment]
yes @seanbudd ,
It is a chat bot application which we have built.
Will share the html code sample for the same issue

**Comment by seanbudd** [comment]
I think you will need to get the html sample from microsoft teams. This issue seems similar to #14291

**Comment by PuspanjaliMuduli** [comment]
Okay, I will try to share the html code for this issue, but if you can help me with the steps to get the html sample it will be helpful. 
Thanks!

**Comment by PuspanjaliMuduli** [comment]
Hello @seanbudd ,

Please find the html sample for this adaptive card design & I have also attached the UI of the card.
[html-sample-14448.txt](https://github.com/nvaccess/nvda/files/10256811/html-sample-14448.txt)
<img width="323" alt="adaptivecard-UI" src="https://user-images.githubusercontent.com/109591184/208370825-895025f4-fb83-46a8-8e5d-f33bc1c54fbe.PNG">

**Comment by seanbudd** [comment]
This needs to be a minimal HTML sample, something that we as developers can debug and understand.

Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/) which reproduces the issue.

**Comment by PuspanjaliMuduli** [comment]
okay, I will try to share the html file again for this issue. Thanks

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd ,

[issue-14448.txt](https://github.com/nvaccess/nvda/files/10299760/issue-14448.txt)
Please download & check, I am not able to send html file format here. So I have sent the html in txt file.
Please check & let me know if you need any more details.

**Comment by seanbudd** [comment]
@PuspanjaliMuduli, this file is quite similar to the previous one.
As mentioned, this sample must be **minimal**.
A minimal example does not contain code that is not necessary in reproducing the issue.
We cannot debug the file you have given.

these resources may explain what a minimal reproducible example is better.
https://en.wikipedia.org/wiki/Minimal_reproducible_example
https://stackoverflow.com/help/minimal-reproducible-example

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd,

Thanks for sharing the example document here.
we will check & try to share minimal html code in some time for this issue.
Thank you!

**Comment by PuspanjaliMuduli** [comment]
Hello @seanbudd, 

PFA for the updated html where we have removed the unwanted code. We have only kept the html where there is adaptive card text.

Please let me know if you need any changes.
[MInimal-html-sample-14448.txt](https://github.com/nvaccess/nvda/files/10341680/MInimal-html-sample-14448.txt)

**Comment by seanbudd** [comment]
Thanks @PuspanjaliMuduli !

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd ,

could you please share if there is any update on this issue.
Thanks!

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd,

could you please share if there is any update on this issue.
Thanks!

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd,

could you please share if there is any update on this issue.
Thanks!

**Comment by PuspanjaliMuduli** [comment]
Hi @seanbudd,

could you please share if there is any update on this issue.
Thanks!

**Comment by seanbudd** [comment]
@PuspanjaliMuduli there are almost 3000 open issues. This is a lower priority issue. When there is an update on the issue, it will be shared here. Please stop spamming. If you want to see this issue fixed sooner, you are welcome to open a pull request.

**Comment by Adriani90** [comment]
This is the text which is repeated by NVDA: "Please enter the full name of the person for refined results. E.g. Matthew Perry".

This seems to be a label inside a span. However, I am not sure why NVDA repeats this twice in your case.

I cannot reproduce this in Microsoft Teams 1.6.0 January 2023 update and NVDA alpha-27769,872db60c (2023.2.0.27769)

Are you still having this problem?

**Comment by PuspanjaliMuduli** [comment]
okay @Adriani90 ,

Thanks for the update, we will check again & let you know if the issue persists.

**Comment by PuspanjaliMuduli** [comment]
Hi @Adriani90,

We are using NVDA version 2022.4(2022.4.7.27401) & Teams version is 1.6.00.1381 (64-bit).
But we are able to reproduce the same issue with NVDA screen reader.
Could you please help here.
Thanks!

**Comment by PuspanjaliMuduli** [comment]
Hi @Adriani90,

Could you please help here to update the NVDA version to alpha-27769,872db60c (2023.2.0.27769).
As we are not getting the options to update to this version, but the issue still exists in NVDA version 2022.4(2022.4.7.27401).
Let us know if you have any more update on this case.
Thanks!

**Comment by Adriani90** [comment]
I think I've managed to reproduce this issue, even with the last alpha of NVDA. This repetition occurs on the message edit field as well, when you send a message to someone. NVDA repeats the text "enter a new message edit field enter a new message" both in focus mode and in browse mode.
So I don't really know what's happening here, why NVDA repeats this. I have removed the label, removed even the aria-label of the div class=ui box but NVDA still repeats it. Maybe you have to play a bit on removing aria-label and the aria-describedby atribute from different div elements and see if NVDA still repeats this.


Note I am not a core developer of NVDA, so I guess @seanbudd can give a more competent answer here. The last test case you provided is good, and this can even be reproduced in Chrome or other browsers, when the setting "automatically changes to focus mode" in browse mode settings is enabled.

**Comment by Adriani90** [comment]
I've seen such repetition of reportings in many cases in form fields, I don't know exactly the root of this double speaking.

**Comment by PuspanjaliMuduli** [comment]
Hi @Adriani90 ,

Thanks for the update.
Hello @seanbudd,
Could you please help here to share the latest update on this.
Thanks!

**Comment by seanbudd** [comment]
@PuspanjaliMuduli there are almost 3000 open issues. This is a lower priority issue. As mentioned before, when there is an update on the issue, it will be shared here. Please stop spamming. If you want to see this issue fixed sooner, you are welcome to open a pull request fixing the issue yourself.
Continuing to bump issues without adding information will result in a temporary ban from the repository.

**Comment by Adriani90** [comment]
This is an implementation issue with MS Teams. MS Teams seems to render the label of a field 2 times:
1. first on the parent level directly on the edit field object (probably aria-label which is not visible), and 
2. a second time on the child object of the edit field itself as a readOnly description (visible on the screen for sighted people).

On the child object level of an adaptive card edit field, typed characters seem to be rendered twice, this is an implementation issue which needs to be fixed by Microsoft.

This causes the reporting of the edit field label twice (e.g. enter a message edit enter a message) when the edit field is blank.
When you type the first letter or character in the edit field, the visible indication disapears, so the label "enter a message" is reported only once.

cc: @msftedad can you take this up with the developers team?

**Comment by msftedad** [comment]
@Adriani90, Since we are unable to reproduce the issue, could you please provide more information on where we can find UNAbot UAT or a similar chatbot that uses adaptive cards?
